From e8d919a6f2b58ecd4e648366dced2668ece241e0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Sun, 24 Feb 2013 17:20:33 +0200
Subject: ARM: i.MX6: cm-fx6: remove csi related board code

Remove both, camera and tvin code.
The CSI code currently is not used on cm-fx6.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |   43 --------
 arch/arm/mach-mx6/board-cm-fx6-q.h  |   43 --------
 arch/arm/mach-mx6/board-cm-fx6.c    |  198 +++--------------------------------
 3 files changed, 13 insertions(+), 271 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index fcacd5a..3dc0449 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -168,49 +168,6 @@ static iomux_v3_cfg_t mx6dl_arm2_can_pads[] = {
 	MX6DL_PAD_CSI0_DAT6__GPIO_5_24,	/* CAN2 EN */
 };
 
-static iomux_v3_cfg_t mx6dl_arm2_csi0_sensor_pads[] = {
-	MX6DL_PAD_GPIO_0__CCM_CLKO,
-	/* ipu1 csi0 */
-	MX6DL_PAD_CSI0_DAT12__IPU1_CSI0_D_12,
-	MX6DL_PAD_CSI0_DAT13__IPU1_CSI0_D_13,
-	MX6DL_PAD_CSI0_DAT14__IPU1_CSI0_D_14,
-	MX6DL_PAD_CSI0_DAT15__IPU1_CSI0_D_15,
-	MX6DL_PAD_CSI0_DAT16__IPU1_CSI0_D_16,
-	MX6DL_PAD_CSI0_DAT17__IPU1_CSI0_D_17,
-	MX6DL_PAD_CSI0_DAT18__IPU1_CSI0_D_18,
-	MX6DL_PAD_CSI0_DAT19__IPU1_CSI0_D_19,
-	MX6DL_PAD_CSI0_VSYNC__IPU1_CSI0_VSYNC,
-	MX6DL_PAD_CSI0_MCLK__IPU1_CSI0_HSYNC,
-	MX6DL_PAD_CSI0_PIXCLK__IPU1_CSI0_PIXCLK,
-	/* camera reset */
-	MX6DL_PAD_GPIO_19__GPIO_4_5,
-	/* camera powerdown */
-	MX6DL_PAD_CSI0_DAT5__GPIO_5_23,
-};
-
-static iomux_v3_cfg_t mx6dl_arm2_csi0_tvin_pads[] = {
-	/* ipu1 csi0 */
-	MX6DL_PAD_CSI0_DAT12__IPU1_CSI0_D_12,
-	MX6DL_PAD_CSI0_DAT13__IPU1_CSI0_D_13,
-	MX6DL_PAD_CSI0_DAT14__IPU1_CSI0_D_14,
-	MX6DL_PAD_CSI0_DAT15__IPU1_CSI0_D_15,
-	MX6DL_PAD_CSI0_DAT16__IPU1_CSI0_D_16,
-	MX6DL_PAD_CSI0_DAT17__IPU1_CSI0_D_17,
-	MX6DL_PAD_CSI0_DAT18__IPU1_CSI0_D_18,
-	MX6DL_PAD_CSI0_DAT19__IPU1_CSI0_D_19,
-	MX6DL_PAD_CSI0_VSYNC__IPU1_CSI0_VSYNC,
-	MX6DL_PAD_CSI0_MCLK__IPU1_CSI0_HSYNC,
-	MX6DL_PAD_CSI0_PIXCLK__IPU1_CSI0_PIXCLK,
-	/* camera reset */
-	MX6DL_PAD_CSI0_DAT7__GPIO_5_25,
-	/* camera powerdown */
-	MX6DL_PAD_CSI0_DAT5__GPIO_5_23,
-};
-
-static iomux_v3_cfg_t mx6dl_arm2_mipi_sensor_pads[] = {
-	MX6DL_PAD_CSI0_MCLK__CCM_CLKO,
-};
-
 #define MX6DL_USDHC_PAD_SETTING(id, speed)	\
 mx6dl_sd##id##_##speed##mhz[] = {		\
 	MX6DL_PAD_SD##id##_CLK__USDHC##id##_CLK_##speed##MHZ,	\
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index dbd00a1..bbf7729 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -168,49 +168,6 @@ static iomux_v3_cfg_t mx6q_arm2_can_pads[] = {
 	MX6Q_PAD_CSI0_DAT6__GPIO_5_24,	/* CAN2 EN */
 };
 
-static iomux_v3_cfg_t mx6q_arm2_csi0_sensor_pads[] = {
-	MX6Q_PAD_GPIO_0__CCM_CLKO,
-	/* ipu1 csi0 */
-	MX6Q_PAD_CSI0_DAT12__IPU1_CSI0_D_12,
-	MX6Q_PAD_CSI0_DAT13__IPU1_CSI0_D_13,
-	MX6Q_PAD_CSI0_DAT14__IPU1_CSI0_D_14,
-	MX6Q_PAD_CSI0_DAT15__IPU1_CSI0_D_15,
-	MX6Q_PAD_CSI0_DAT16__IPU1_CSI0_D_16,
-	MX6Q_PAD_CSI0_DAT17__IPU1_CSI0_D_17,
-	MX6Q_PAD_CSI0_DAT18__IPU1_CSI0_D_18,
-	MX6Q_PAD_CSI0_DAT19__IPU1_CSI0_D_19,
-	MX6Q_PAD_CSI0_VSYNC__IPU1_CSI0_VSYNC,
-	MX6Q_PAD_CSI0_MCLK__IPU1_CSI0_HSYNC,
-	MX6Q_PAD_CSI0_PIXCLK__IPU1_CSI0_PIXCLK,
-	/* camera reset */
-	MX6Q_PAD_GPIO_19__GPIO_4_5,
-	/* camera powerdown */
-	MX6Q_PAD_CSI0_DAT5__GPIO_5_23,
-};
-
-static iomux_v3_cfg_t mx6q_arm2_csi0_tvin_pads[] = {
-	/* ipu1 csi0 */
-	MX6Q_PAD_CSI0_DAT12__IPU1_CSI0_D_12,
-	MX6Q_PAD_CSI0_DAT13__IPU1_CSI0_D_13,
-	MX6Q_PAD_CSI0_DAT14__IPU1_CSI0_D_14,
-	MX6Q_PAD_CSI0_DAT15__IPU1_CSI0_D_15,
-	MX6Q_PAD_CSI0_DAT16__IPU1_CSI0_D_16,
-	MX6Q_PAD_CSI0_DAT17__IPU1_CSI0_D_17,
-	MX6Q_PAD_CSI0_DAT18__IPU1_CSI0_D_18,
-	MX6Q_PAD_CSI0_DAT19__IPU1_CSI0_D_19,
-	MX6Q_PAD_CSI0_VSYNC__IPU1_CSI0_VSYNC,
-	MX6Q_PAD_CSI0_MCLK__IPU1_CSI0_HSYNC,
-	MX6Q_PAD_CSI0_PIXCLK__IPU1_CSI0_PIXCLK,
-	/* camera reset */
-	MX6Q_PAD_CSI0_DAT7__GPIO_5_25,
-	/* camera powerdown */
-	MX6Q_PAD_CSI0_DAT5__GPIO_5_23,
-};
-
-static iomux_v3_cfg_t mx6q_arm2_mipi_sensor_pads[] = {
-	MX6Q_PAD_CSI0_MCLK__CCM_CLKO,
-};
-
 #define MX6Q_USDHC_PAD_SETTING(id, speed)	\
 mx6q_sd##id##_##speed##mhz[] = {		\
 	MX6Q_PAD_SD##id##_CLK__USDHC##id##_CLK_##speed##MHZ,	\
diff --git a/arch/arm/mach-mx6/board-cm-fx6.c b/arch/arm/mach-mx6/board-cm-fx6.c
index 40dd526..737d00f 100644
--- a/arch/arm/mach-mx6/board-cm-fx6.c
+++ b/arch/arm/mach-mx6/board-cm-fx6.c
@@ -60,7 +60,6 @@
 #include <mach/mxc_hdmi.h>
 #include <mach/mxc_asrc.h>
 #include <mach/mipi_dsi.h>
-#include <mach/mipi_csi2.h>
 
 #include <asm/irq.h>
 #include <asm/setup.h>
@@ -85,11 +84,8 @@
 #define MX6_ARM2_DISP0_I2C_EN		IMX_GPIO_NR(3, 28)
 #define MX6_ARM2_CAP_TCH_INT		IMX_GPIO_NR(3, 31)
 #define MX6_ARM2_DISP0_DET_INT		IMX_GPIO_NR(3, 31)
-#define MX6_ARM2_CSI0_RST		IMX_GPIO_NR(4, 5)
 #define MX6_ARM2_DISP0_RESET		IMX_GPIO_NR(5, 0)
-#define MX6_ARM2_CSI0_PWN		IMX_GPIO_NR(5, 23)
 #define MX6_ARM2_CAN2_EN		IMX_GPIO_NR(5, 24)
-#define MX6_ARM2_CSI0_RST_TVIN		IMX_GPIO_NR(5, 25)
 #define MX6_ARM2_SD3_CD			IMX_GPIO_NR(6, 11)
 #define MX6_ARM2_SD3_WP			IMX_GPIO_NR(6, 14)
 #define MX6_ARM2_CAN1_STBY		IMX_GPIO_NR(7, 12)
@@ -370,129 +366,6 @@ static struct fsl_mxc_dvi_platform_data sabr_ddc_dvi_data = {
 	.update		= ddc_dvi_update,
 };
 
-static void mx6_csi0_io_init(void)
-{
-	iomux_v3_cfg_t *sensor_pads = NULL;
-	u32 sensor_pads_cnt;
-
-	if (cpu_is_mx6q()) {
-		sensor_pads = mx6q_arm2_csi0_sensor_pads;
-		sensor_pads_cnt = ARRAY_SIZE(mx6q_arm2_csi0_sensor_pads);
-	} else if (cpu_is_mx6dl()) {
-		sensor_pads = mx6dl_arm2_csi0_sensor_pads;
-		sensor_pads_cnt = ARRAY_SIZE(mx6dl_arm2_csi0_sensor_pads);
-	}
-
-	BUG_ON(!sensor_pads);
-	mxc_iomux_v3_setup_multiple_pads(sensor_pads, sensor_pads_cnt);
-
-	/* Camera reset */
-	gpio_request(MX6_ARM2_CSI0_RST, "cam-reset");
-	gpio_direction_output(MX6_ARM2_CSI0_RST, 1);
-
-	/* Camera power down */
-	gpio_request(MX6_ARM2_CSI0_PWN, "cam-pwdn");
-	gpio_direction_output(MX6_ARM2_CSI0_PWN, 1);
-	msleep(1);
-	gpio_set_value(MX6_ARM2_CSI0_PWN, 0);
-
-	/* For MX6Q:
-	 * GPR1 bit19 and bit20 meaning:
-	 * Bit19:       0 - Enable mipi to IPU1 CSI0
-	 *                      virtual channel is fixed to 0
-	 *              1 - Enable parallel interface to IPU1 CSI0
-	 * Bit20:       0 - Enable mipi to IPU2 CSI1
-	 *                      virtual channel is fixed to 3
-	 *              1 - Enable parallel interface to IPU2 CSI1
-	 * IPU1 CSI1 directly connect to mipi csi2,
-	 *      virtual channel is fixed to 1
-	 * IPU2 CSI0 directly connect to mipi csi2,
-	 *      virtual channel is fixed to 2
-	 *
-	 * For MX6DL:
-	 * GPR1 bit 21 and GPR13 bit 0-5, RM has detail information
-	 */
-	if (cpu_is_mx6q())
-		mxc_iomux_set_gpr_register(1, 19, 1, 1);
-	else if (cpu_is_mx6dl())
-		mxc_iomux_set_gpr_register(13, 0, 3, 4);
-}
-
-static struct fsl_mxc_camera_platform_data camera_data = {
-	.analog_regulator	= "DA9052_LDO7",
-	.core_regulator		= "DA9052_LDO9",
-	.mclk			= 24000000,
-	.mclk_source = 0,
-	.csi			= 0,
-	.io_init		= mx6_csi0_io_init,
-};
-
-static void mx6_csi0_tvin_io_init(void)
-{
-	iomux_v3_cfg_t *tvin_pads = NULL;
-	u32 tvin_pads_cnt;
-
-	if (cpu_is_mx6q()) {
-		tvin_pads = mx6q_arm2_csi0_tvin_pads;
-		tvin_pads_cnt = ARRAY_SIZE(mx6q_arm2_csi0_tvin_pads);
-	} else if (cpu_is_mx6dl()) {
-		tvin_pads = mx6dl_arm2_csi0_tvin_pads;
-		tvin_pads_cnt = ARRAY_SIZE(mx6dl_arm2_csi0_tvin_pads);
-	}
-
-	BUG_ON(!tvin_pads);
-	mxc_iomux_v3_setup_multiple_pads(tvin_pads, tvin_pads_cnt);
-
-	/* Tvin reset */
-	gpio_request(MX6_ARM2_CSI0_RST_TVIN, "tvin-reset");
-	gpio_direction_output(MX6_ARM2_CSI0_RST_TVIN, 1);
-
-	/* Tvin power down */
-	gpio_request(MX6_ARM2_CSI0_PWN, "cam-pwdn");
-	gpio_direction_output(MX6_ARM2_CSI0_PWN, 0);
-	msleep(1);
-	gpio_set_value(MX6_ARM2_CSI0_PWN, 1);
-
-	if (cpu_is_mx6q())
-		mxc_iomux_set_gpr_register(1, 19, 1, 1);
-	else if (cpu_is_mx6dl())
-		mxc_iomux_set_gpr_register(13, 0, 3, 4);
-}
-
-static struct fsl_mxc_tvin_platform_data tvin_data = {
-	.io_init = mx6_csi0_tvin_io_init,
-	.cvbs = false,
-};
-
-static void mx6_mipi_sensor_io_init(void)
-{
-	iomux_v3_cfg_t *mipi_sensor_pads = NULL;
-	u32 mipi_sensor_pads_cnt;
-
-	if (cpu_is_mx6q()) {
-		mipi_sensor_pads = mx6q_arm2_mipi_sensor_pads;
-		mipi_sensor_pads_cnt = ARRAY_SIZE(mx6q_arm2_mipi_sensor_pads);
-	} else if (cpu_is_mx6dl()) {
-		mipi_sensor_pads = mx6dl_arm2_mipi_sensor_pads;
-		mipi_sensor_pads_cnt = ARRAY_SIZE(mx6dl_arm2_mipi_sensor_pads);
-
-	}
-	BUG_ON(!mipi_sensor_pads);
-	mxc_iomux_v3_setup_multiple_pads(mipi_sensor_pads,
-					mipi_sensor_pads_cnt);
-
-	/*for mx6dl, mipi virtual channel 1 connect to csi 1*/
-	if (cpu_is_mx6dl())
-		mxc_iomux_set_gpr_register(13, 3, 3, 1);
-}
-
-static struct fsl_mxc_camera_platform_data ov5640_mipi_data = {
-	.mclk		= 24000000,
-	.csi		= 1,
-	.mclk_source = 0,
-	.io_init	= mx6_mipi_sensor_io_init,
-};
-
 static int sii902x_get_pins(void)
 {
 	/* Sii902x HDMI controller */
@@ -525,16 +398,6 @@ static struct fsl_mxc_lcd_platform_data sii902x_hdmi_data = {
 	.put_pins = sii902x_put_pins,
 };
 
-static struct i2c_board_info mxc_i2c0_board_info[] __initdata = {
-	{
-		I2C_BOARD_INFO("ov5640", 0x3c),
-		.platform_data = (void *)&camera_data,
-	}, {
-		I2C_BOARD_INFO("adv7180", 0x21),
-		.platform_data = (void *)&tvin_data,
-	},
-};
-
 static struct imxi2c_platform_data mx6_arm2_i2c0_data = {
 	.bitrate = 100000,
 };
@@ -547,30 +410,24 @@ static struct imxi2c_platform_data mx6_arm2_i2c2_data = {
 	.bitrate = 400000,
 };
 
-static struct i2c_board_info mxc_i2c2_board_info[] __initdata = {
+static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
 	{
-		I2C_BOARD_INFO("mxc_dvi", 0x50),
-		.platform_data = &sabr_ddc_dvi_data,
-		.irq = gpio_to_irq(MX6_ARM2_DISP0_DET_INT),
-	}, {
-		I2C_BOARD_INFO("egalax_ts", 0x4),
-		.irq = gpio_to_irq(MX6_ARM2_CAP_TCH_INT),
-	}, {
-		I2C_BOARD_INFO("sii902x", 0x39),
-		.platform_data = &sii902x_hdmi_data,
-		.irq = gpio_to_irq(MX6_ARM2_DISP0_DET_INT),
+//		I2C_BOARD_INFO("mxc_dvi", 0x50),
+//		.platform_data = &sabr_ddc_dvi_data,
+//		.irq = gpio_to_irq(MX6_ARM2_DISP0_DET_INT),
+		I2C_BOARD_INFO("mxc_hdmi_i2c", 0x50),
+//	}, {
+//		I2C_BOARD_INFO("egalax_ts", 0x4),
+//		.irq = gpio_to_irq(MX6_ARM2_CAP_TCH_INT),
+//	}, {
+//		I2C_BOARD_INFO("sii902x", 0x39),
+//		.platform_data = &sii902x_hdmi_data,
+//		.irq = gpio_to_irq(MX6_ARM2_DISP0_DET_INT),
 	},
 };
 
-static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
+static struct i2c_board_info mxc_i2c2_board_info[] __initdata = {
 	{
-		I2C_BOARD_INFO("egalax_ts", 0x4),
-		.irq = gpio_to_irq(MX6_ARM2_CAP_TCH_INT),
-	}, {
-		I2C_BOARD_INFO("mxc_hdmi_i2c", 0x50),
-	}, {
-		I2C_BOARD_INFO("ov5640_mipi", 0x3c),
-		.platform_data = (void *)&ov5640_mipi_data,
 	},
 };
 
@@ -877,30 +734,6 @@ static const struct flexcan_platform_data
 	}
 };
 
-static struct mipi_csi2_platform_data mipi_csi2_pdata = {
-	.ipu_id		= 0,
-	.csi_id		= 0,
-	.v_channel	= 0,
-	.lanes		= 2,
-	.dphy_clk	= "mipi_pllref_clk",
-	.pixel_clk	= "emi_clk",
-};
-
-static struct fsl_mxc_capture_platform_data capture_data[] = {
-	{
-		.csi = 0,
-		.ipu = 0,
-		.mclk_source = 0,
-		.is_mipi = 0,
-	}, {
-		.csi = 1,
-		.ipu = 0,
-		.mclk_source = 0,
-		.is_mipi = 1,
-	},
-};
-
-
 static void arm2_suspend_enter(void)
 {
 	/* suspend preparation */
@@ -1143,7 +976,6 @@ static void __init cm_fx6_init(void)
 	pu_reg_id = arm2_dvfscore_data.pu_id;
 	mx6_arm2_init_uart();
 
-	imx6q_add_mipi_csi2(&mipi_csi2_pdata);
 	imx6q_add_mxc_hdmi_core(&hdmi_core_data);
 
 	imx6q_add_ipuv3(0, &ipu_data[0]);
@@ -1168,8 +1000,6 @@ static void __init cm_fx6_init(void)
 	imx6q_add_lcdif(&lcdif_data);
 	imx6q_add_ldb(&ldb_data);
 	imx6q_add_v4l2_output(0);
-	imx6q_add_v4l2_capture(0, &capture_data[0]);
-	imx6q_add_v4l2_capture(1, &capture_data[1]);
 
 	imx6q_add_imx_snvs_rtc();
 
@@ -1177,8 +1007,6 @@ static void __init cm_fx6_init(void)
 
 	imx6q_add_imx_i2c(0, &mx6_arm2_i2c0_data);
 	imx6q_add_imx_i2c(1, &mx6_arm2_i2c1_data);
-	i2c_register_board_info(0, mxc_i2c0_board_info,
-			ARRAY_SIZE(mxc_i2c0_board_info));
 	i2c_register_board_info(1, mxc_i2c1_board_info,
 			ARRAY_SIZE(mxc_i2c1_board_info));
 	if (!spdif_en) {
-- 
1.7.9.5

